sdk/communication/Azure.ResourceManager.Communication/tests/ScenarioTests/SuppressionListTests.cs

```diff
@@ -0,0 +1,211 @@
+
+                Assert.IsNotNull(resource);
+                Assert.AreEqual(resource.Data.ListName, resourceName.Value);
+            }
+        }
+
+        [Test]
+        public async Task AddAddresses()
+        {
+            var listName = "addressList";
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+
+            Assert.IsNotNull(suppressionList);
+
+            var addresses = new string[] { "user1@email.com", "user2@email.com", "user3@email.com" };
+            var resourceNames = new Dictionary<string, string>();
+
+            foreach (var address in addresses)
+            {
+                var resource = await CreateDefaultSuppressionListResource(_domainResource, address);
+                resourceNames[resource.Data.Name] = address;
+            }
+
+            var suppressionListAddresses = await suppressionList.GetSuppressionListAddressResources().GetAllAsync().ToEnumerableAsync();
+
+            Assert.IsNotNull(suppressionListAddresses);
+            Assert.IsTrue(suppressionListAddresses.Count() >= addresses.Length);
+
+            foreach (var resourceName in resourceNames)
+            {
+                var resource = suppressionListAddresses.Where(s => s.Data.Name == resourceName.Key).FirstOrDefault();
+
+                Assert.IsNotNull(resource);
+                Assert.AreEqual(resource.Data.Email, resourceName.Value);
+            }
+        }
+
+        [Test]
+        public async Task UpdateAddress()
+        {
+            var listName = "addressList";
+            var suppressionList = await CreateDefaultSuppressionListResource(_domainResource, listName);
+
+            Assert.IsNotNull(suppressionList);
+
+            var email = "superuser@email.com";
+            var created = await CreateDefaultSuppressionListAddressResource(suppressionList, email, "firstName", "lastName", "notes");
+
+            SuppressionListAddressResourceData data = new SuppressionListAddressResourceData
+            {
+                Email = created.Data.Email,
+                FirstName = $"updated+{created.Data.FirstName}",
+                LastName = $"updated+{created.Data.LastName}",
+                Notes = created.Data.Notes
+            };
+
+            var updated = await suppressionList
+                .GetSuppressionListAddressResources()
+                .CreateOrUpdateAsync(WaitUntil.Completed, created.Id.Name, data);
+
+            Assert.IsNotNull(updated);
+            Assert.AreEqual(created.Data.Id, updated.Value.Data.Id);
+            Assert.AreEqual(created.Data.Email, updated.Value.Data.Email);
+            Assert.AreNotEqual(created.Data.FirstName, updated.Value.Data.FirstName);
+            Assert.AreNotEqual(created.Data.LastName, updated.Value.Data.LastName);
+            Assert.AreEqual(created.Data.Notes, updated.Value.Data.Notes);
+        }
```

Comment:
Can you add one check to check for the total count of the list item after remains the same as before?